instance/models/instance.py

```diff
@@ -66,13 +68,104 @@ class InconsistentInstanceState(Exception):
     """
     pass
 
+
+# States ######################################################################
+
+class InstanceState(ResourceState):
+    """
+    A [finite state machine] state describing an instance.
+    """
+    # A state is "steady" if we don't expect it to change.
+    # This information can be used to:
+    # - delay execution of an operation until the target server reaches a steady state
+    # - raise an exception when trying to schedule an operation that depends on a state change
+    #   while the target server is in a steady state
+    # Steady states include: Status.Running, Status.ConfigurationFailed, Status.Error, Status.Terminated
```

Comment:
Shouldn't `New` also be a steady state? Because I can instantiate an instance, and it will have status `New`, but it will stay in that status forever until we call `provision_instance()`.

This would also fix an issue in the UI, where if you create an instance but don't call `provision_instance`, there is no option to provision it seen in the UI, and it looks like something is in progress, when in reality nothing is happening.